Test/Mocks/ClientMock.php

```diff
@@ -0,0 +1,259 @@
+        BackupRateOriginAddress $backupRateOriginAddress
+    ) {
+        $this->tjHelper = $tjHelper;
+        $this->taxjarConfig = $taxjarConfig;
+        $this->backupRateOriginAddress = $backupRateOriginAddress;
+        $this->apiKey = 'test-api-key';
+    }
+
+    /**
+     * Perform a GET request
+     *
+     * @param string $resource
+     * @param array $errors
+     * @return array
+     */
+    public function getResource($resource, $errors = [])
+    {
+        return $this->mockResponse;
+//        $client = $this->getClient($this->_getApiUrl($resource));
+//        return $this->_getRequest($client, $errors);
+    }
+
+    /**
+     * Perform a POST request
+     *
+     * @param string $resource
+     * @param array $data
+     * @param array $errors
+     * @return array
+     */
+    public function postResource($resource, $data, $errors = [])
+    {
+        $client = $this->getClient($this->_getApiUrl($resource), \Zend_Http_Client::POST);
+        $client->setRawData(json_encode($data), 'application/json');
+        return $this->_getRequest($client, $errors);
+    }
+
+    /**
+     * Perform a PUT request
+     *
+     * @param string $resource
+     * @param int $resourceId
+     * @param array $data
+     * @param array $errors
+     * @return array
+     */
+    public function putResource($resource, $resourceId, $data, $errors = [])
+    {
+        $resourceUrl = $this->_getApiUrl($resource) . '/' . $resourceId;
+        $client = $this->getClient($resourceUrl, \Zend_Http_Client::PUT);
+        $client->setRawData(json_encode($data), 'application/json');
+        return $this->_getRequest($client, $errors);
+    }
+
+    /**
+     * Perform a DELETE request
+     *
+     * @param string $resource
+     * @param int $resourceId
+     * @param array $errors
+     * @return array
+     */
+    public function deleteResource($resource, $resourceId, $errors = [])
+    {
+        $resourceUrl = $this->_getApiUrl($resource) . '/' . $resourceId;
+        $client = $this->getClient($resourceUrl, \Zend_Http_Client::DELETE);
+        return $this->_getRequest($client, $errors);
+    }
+
+    /**
+     * Set API token for client requests
+     *
+     * @param string $key
+     * @return void
+     */
+    public function setApiKey($key)
+    {
+        $this->apiKey = $key;
+    }
+
+    /**
+     * @param bool $toggle
+     * @return void
+     */
+    public function showResponseErrors($toggle)
+    {
+        $this->showResponseErrors = $toggle;
+    }
+
+    /**
+     * Get HTTP Client
+     *
+     * @param string $url
+     * @param string $method
+     * @return \Zend_Http_Client $client
+     */
+    private function getClient($url, $method = \Zend_Http_Client::GET)
+    {
+        // @codingStandardsIgnoreStart
+        $client = new \Zend_Http_Client($url, ['timeout' => 30]);
+        // @codingStandardsIgnoreEnd
+        $client->setUri($url);
+        $client->setMethod($method);
+//        $client->setConfig([
```

Comment:
Theres some commented out code here that should probably be cleaned up. Also this is one of those private methods I mentioned in an earlier comment that I don't think are necessary for a mock class.